doc/TockBinaryFormat.md

```diff
@@ -17,27 +17,41 @@
     + [`6` Permissions](#6-permissions)
     + [`7` Persistent ACL](#7-persistent-acl)
     + [`8` Kernel Version](#8-kernel-version)
+    + [`9` Program](#9-program)
+    + [`128` Credentials Footer](#128-credentials-footer)
 - [Code](#code)
 
 <!-- tocstop -->
 
-Tock process binaries are must be in the Tock Binary Format (TBF). A TBF
-includes a header portion, which encodes meta-data about the process, followed
-by a binary blob which is executed directly, followed by optional padding.
+Tock userspace applications must following the Tock Binary Format
+(TBF). A TBF Object has four parts: a Header section, which encodes
+meta-data about the TBF Object, the actual Userspace Binary, an
+optional Footer section which encodes metadata about the TBF Object,
+and finally optional padding. 
+
+TBF Headers and Footers differ in how they are handled for TBF Object
+integrity. Integrity values (e.g., hashes) for a TBF Object are
+computed over the Header section and Userspace Binary but not the
+Footer Region or padding. TBF Headers are covered by integrity, while
```

Comment:
```suggestion
Footer Region or padding after footers. TBF Headers are covered by integrity, while
```

There are two padding sections in the diagram below, make this slightly more explicit.